File: languages/go/entity/identity/decimal_attribute_descriptor.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// DecimalAttributeDescriptor represents a descriptor for an attribute with a decimal value.
type DecimalAttributeDescriptor struct {
	AttributeDescriptor
}

// NewDecimalAttributeDescriptor creates a new DecimalAttributeDescriptor instance with the provided details.
func NewDecimalAttributeDescriptor(name string, id string, description string, required bool) DecimalAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return DecimalAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewDecimalAttributeDescriptorFromProto(s *proto.DecimalAttributeDefinition) DecimalAttributeDescriptor {
	if s == nil {
		return DecimalAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return DecimalAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s DecimalAttributeDescriptor) ToProto() *proto.DecimalAttributeDefinition {
	return &proto.DecimalAttributeDefinition{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}

File: languages/js/src/entity/identity/decimal_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/bloock_identity_entities";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for an attribute with a decimal value.
 */
export class DecimalAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.DecimalAttributeDefinition {
    return identityEntitiesProto.DecimalAttributeDefinition.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.DecimalAttributeDefinition
  ): DecimalAttributeDescriptor {
    return new DecimalAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}
